code/handlers/trackservers.q

```diff
@@ -175,12 +173,15 @@ retrydiscovery:{
 // Called by the discovery service when it restarts
 autodiscovery:{if[DISCOVERYRETRY>0; .servers.retrydiscovery[]]}
 
+//check if .proc.getattributes exists in the nontorqprocess
+checknontorqattr:{$[(`proc in key`)&(first distinct`getattributes`cd in key`.proc);1;0]};
+
 // Attempt to make a connection for specified row ids
 retryrows:{[rows]
     // opencon, amends global tables, cannot be used inside of a select statement
     handles:.servers.opencon each exec hpup from`.servers.SERVERS where i in rows;
-    update lastp:.proc.cp[],w:handles from`.servers.SERVERS where i in rows;
-        update attributes:{$[null x;()!();@[x;(`.proc.getattributes;`);()!()]]} each w,startp:?[null w;0Np;.proc.cp[]] from `.servers.SERVERS where i in rows;
+    update lastp:.proc.cp[],w:handles from`.servers.SERVERS where i in rows; 
+        update attributes:{$[null x;()!();$[@[x;(checknontorqattr;())];@[x;(`.proc.getattributes;`);()!()];()!()]]} each w,startp:?[null w;0Np;.proc.cp[]] from `.servers.SERVERS where i in rows;
```

Comment:
this line is a little unwieldy... nested `$` conditional can probably be simplified a bit?

At the very least, can get rid of inner `$` and use the 'if...else if...else' `$[cond1;val1;cond2;val2;val3]` style

Reply:
Will change it now.